### PR TITLE
[FIX] project,sale_timesheet: fix profitability section visibility

### PR DESCRIPTION
**Pull request by rish-odoo**
Steps:
- Install project & accounting
- Open project
- Open project.update
- There is profitability section
- Go to accounting module configuration,settings
- Analytic accounting field true or false

Issue:
- Analytic accounting feature true or false the profitability section always visible in project.update

Cause:
- There is no condition for profitability section visibility

Fix:
- Added  required conditions for profitability section visibility

Task-3484413

### CODE REVIEW

File: addons/project/models/project.py [hunk 1]
```diff
@@ -748,6 +748,7 @@ def get_panel_data(self):
     def _get_user_values(self):
         return {
             'is_project_user': self.user_has_groups('project.group_project_user'),
+            'is_analytic_account':self.user_has_groups('analytic.group_analytic_accounting')
```
Comment (auon-odoo) [inline addons/project/models/project.py:751]:
```suggestion
            'is_analytic_account_user':self.user_has_groups('analytic.group_analytic_accounting')
```

**Review by auon-odoo — CHANGES_REQUESTED** [review]
Thanks for the fix!

**Review by auon-odoo — APPROVED** [review]
Thanks for your work!

**Review by xavierbol — COMMENTED** [review]
Why not simply doing that: 
```suggestion
diff --git a/addons/sale_timesheet/models/project.py b/addons/sale_timesheet/models/project.py
index 8b861897b93a..ef742a72becb 100644
--- a/addons/sale_timesheet/models/project.py
+++ b/addons/sale_timesheet/models/project.py
@@ -377,7 +377,7 @@ class Project(models.Model):
         return super()._get_sale_order_items_query(domain_per_model)

     def _get_profitability_items(self):
-        if not self.user_has_groups('project.group_project_manager'):
+        if not (self.user_has_groups('project.group_project_manager') and self.user_has_groups('analytic.group_analytic_accounting')):
             return {'data': []}
         data = []
         if self.allow_billable:
```
? 🧐 

Btw, I will discuss with Alexandra about that fix because for me, it is a bit late.

**Comment by xavierbol** [comment]
> Why not simply doing that:
> 
> ```
> diff --git a/addons/sale_timesheet/models/project.py b/addons/sale_timesheet/models/project.py
> index 8b861897b93a..ef742a72becb 100644
> --- a/addons/sale_timesheet/models/project.py
> +++ b/addons/sale_timesheet/models/project.py
> @@ -377,7 +377,7 @@ class Project(models.Model):
>          return super()._get_sale_order_items_query(domain_per_model)
> 
>      def _get_profitability_items(self):
> -        if not self.user_has_groups('project.group_project_manager'):
> +        if not (self.user_has_groups('project.group_project_manager') and self.user_has_groups('analytic.group_analytic_accounting')):
>              return {'data': []}
>          data = []
>          if self.allow_billable:
> ```
> 
> ? 🧐
> 
> Btw, I will discuss with Alexandra about that fix because for me, it is a bit late.

Cannot we simply do that instead of your fix?

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 2]
```diff
@@ -2,6 +2,10 @@
 <odoo>
     <data noupdate="1">
 
+        <record id="base.user_admin" model="res.users">
+           <field name="groups_id" eval="[(4, ref('analytic.group_analytic_accounting'))]"/>
+        </record>
```
Comment (xavierbol) [inline addons/sale_timesheet/data/sale_service_demo.xml:7]: It does not seem to be part of this bug fix, right?

Comment (rish-odoo) [reply addons/sale_timesheet/data/sale_service_demo.xml:7]: I added this demo data in order to properly work some profitability related tour and test tour
Also it'll ease the testing for testing I guess 😅

Comment (xavierbol) [reply addons/sale_timesheet/data/sale_service_demo.xml:7]: You changed the functional behavior in the demo data. Before doing that you should ask to Alexandra or change the tests according to your changes

**Review by xavierbol — COMMENTED** [review]
robodoo r+

**Comment by xavierbol** [comment]
robodoo r+

**Comment by xavierbol** [comment]
robodoo r+

**Pull request closed without merging**